### PR TITLE
Remove Python 2 dependencies from developer packages 

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Removes the Python 2 packages from the rpm/deb mantid-developer packages.

**To test:**

Code review.
To be thorough, build the packages locally and test they install correctly.

Refs #27722

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
So I've just took this opportunity to check it on a completely fresh machine, skipping installing previous `mantid-developer packages` and having some issues with the CMake configure step:
```
CMake Error at buildconfig/CMake/PyQtFindImpl.cmake:61 (message):
  Error encountered while determining PyQt confguration:

  Traceback (most recent call last):

    File "/home/david/mantid/buildconfig/CMake/FindPyQt.py", line 108, in <module>
      sys.exit(main())
    File "/home/david/mantid/buildconfig/CMake/FindPyQt.py", line 98, in main
      print(PyQtConfig('PyQt%d' % args.version))
    File "/home/david/mantid/buildconfig/CMake/FindPyQt.py", line 69, in __init__
      "Tried following locations: {}".format(name, pprint.pformat(possible_sip_dirs)))

  RuntimeError: Unable to find PyQt4.  Tried following locations:
  ['/usr/share/python36-sip/PyQt4',

   '/usr/share/python3-sip/PyQt4',
   '/usr/share/sip/PyQt4']
```


Would it be worth bumping our version to 2.0, as I would consider dropping Python 2 a major change?

**Comment by martyngigg** [comment]
Thanks for looking at this. Was this Ubuntu or CentOS?

**Comment by DavidFair** [comment]
Ubuntu sorry, haven't checked Centos yet

**Comment by martyngigg** [comment]
I see what has happened here. The development files are only installed with the Python 2 version: `python-qt4-dev` so we'll have to keep installing that.

**Comment by martyngigg** [comment]
@DavidFair The last push should have fixed it. The CentOS looks okay.

**Comment by DavidFair** [comment]
Yeah, it's fixed it for Ubuntu thanks
Just checking centos locally now

**Review by DavidFair — APPROVED** [review]
This works on Centos/RHEL too

File: buildconfig/dev-packages/deb/mantid-developer/ns-control [hunk 1]
```diff
@@ -3,15 +3,17 @@ Priority: optional
...
+Version: 2.0
 Maintainer: Mantid Project <mantid-tech@mantidproject.org>
 Priority: optional
 Architecture: all
 Depends: git,
   g++,
   clang-format-6.0,
-  cmake-qt-gui(>=3.5),
-  ninja-build,
+  cmake-qt-gui(>=3.13),
```
Comment (gvardany) [inline buildconfig/dev-packages/deb/mantid-developer/ns-control]: This cmake comes from some other repository right? Is this documented somewhere? I had to scroll back in slack channels to find that...

File: buildconfig/dev-packages/deb/mantid-developer/ns-control [hunk 2]
```diff
@@ -38,34 +40,15 @@ Depends: git,
...
-  texlive-latex-extra,
-  dvipng,
-  graphviz,
-  doxygen,
+  libpython3-dev,
+  ninja-build,
   python3-setuptools,
   python3-sip-dev,
   python3-pyqt4,
+  python-qt4-dev,
```
Comment (gvardany) [inline buildconfig/dev-packages/deb/mantid-developer/ns-control]: This seems python 2?

File: buildconfig/dev-packages/deb/mantid-developer/ns-control [hunk 3]
```diff
@@ -38,34 +40,15 @@ Depends: git,
...
   python3-pyqt5,
+  pyqt5-dev,
```
Comment (gvardany) [inline buildconfig/dev-packages/deb/mantid-developer/ns-control]: These two as well.

File: buildconfig/dev-packages/deb/mantid-developer/ns-control (hunk 1 above)
Comment (martyngigg) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: Yep. David added it to the developer instructions: http://developer.mantidproject.org/GettingStarted.html#ubuntu

File: buildconfig/dev-packages/deb/mantid-developer/ns-control (hunk 2 above)
Comment (martyngigg) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: This is actually a confusion in naming. You're right in that most packages named `python-` are Python 2 but `python-qt4-dev` is actually Python version agnostic and just contains `.sip` files that are compatible with both versions.

File: buildconfig/dev-packages/deb/mantid-developer/ns-control (hunk 3 above)
Comment (martyngigg) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: Same for this as `python-qt4-dev`.

File: buildconfig/dev-packages/deb/mantid-developer/ns-control (hunk 2 above)
Comment (gvardany) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: I see. Is there a way to put a comment in this file?

Comment (martyngigg) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: We can have comments in the file just not within the dependency list. Is a comment that's not near the line much use do you think?

Comment (gvardany) [reply buildconfig/dev-packages/deb/mantid-developer/ns-control]: No I don't think it will be useful then.

**Pull request merged**